sdk/identity/identity/package.json

```diff
@@ -80,7 +80,7 @@
   "homepage": "https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/identity/identity/README.md",
   "sideEffects": false,
   "dependencies": {
-    "@azure/core-http": "^1.2.4",
+    "@azure/core-http": "^2.0.0",
```

Comment:
Do you have to update the version of core-http? Seems like an unnecessary risk for a hotfix unless we _have_ to take it

Reply:
I think its because he needs the fix in https://github.com/Azure/azure-sdk-for-js/pull/15906

Reply:
On that note, should this hotfix branch take the update to core-tracing as well?

Reply:
Nah, we can do that when core-tracing goes GA.

Reply:
I still don't think this is the right thing to do. I worry about surprising someone with a whole host of changes unintentionally. Instead (and I know it's more painful, so sorry!), I would recommend releasing a hotfix to core-http with _just_ the fix you need, then using that as the core-http version to release the hotfix to identity

Reply:
After investigating about the new types, doing some simple tests and discussing this with @chradek , it seems safe for us to upgrade to core-http 2.0.0 on this Identity release.

Reply:
The reason I believe it is safe is because
- `@azure/identity` only re-exports `PipelineOptions` for `@azure/core-http`
- `PipelineOptions` doesn't include any enums (doesn't appear to include them directly or transitively)
- Other than transpiling core-http to ES2017, there weren't any breaking changes to the interfaces.
- We're bumping identity to 1.4.0 because we're dropping node 8 support, which means ES2017 should be safe to use.

@maorleger is there anything in particular that worries you? Any testing we could do to assuage your concerns?

Reply:
I'm fine with going to 1.4.0 with this, but was concerned when going to 1.3.1 👍 thanks for verifying the changes are safe